WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/CustomerSearchUICommand.swift

```diff
@@ -65,23 +95,53 @@ final class CustomerSearchUICommand: SearchUICommand {
 
     func synchronizeModels(siteID: Int64, keyword: String, pageNumber: Int, pageSize: Int, onCompletion: ((Bool) -> Void)?) {
         analytics.track(.orderCreationCustomerSearch)
-        let action = CustomerAction.searchCustomers(siteID: siteID, keyword: keyword) { result in
+
+        let action: CustomerAction
+        if featureFlagService.isFeatureFlagEnabled(.betterCustomerSelectionInOrder),
+           keyword.isEmpty {
+            action = synchronizeAllLightCustomersDataAction(siteID: siteID, pageNumber: pageNumber, pageSize: pageSize, onCompletion: onCompletion)
+        } else {
+            action = searchCustomersAction(siteID: siteID, keyword: keyword, pageNumber: pageNumber, pageSize: pageSize, onCompletion: onCompletion)
+        }
+
+        stores.dispatch(action)
+    }
+
+    func didSelectSearchResult(model: Customer, from viewController: UIViewController, reloadData: () -> Void, updateActionButton: () -> Void) {
+        onDidSelectSearchResult(model)
+    }
+
+    func searchResultsPredicate(keyword: String) -> NSPredicate? {
+        guard featureFlagService.isFeatureFlagEnabled(.betterCustomerSelectionInOrder),
+              keyword.isEmpty else {
+            return NSPredicate(format: "siteID == %lld AND ANY searchResults.keyword = %@", siteID, keyword)
+        }
+
+        return nil
+    }
+}
+
+private extension CustomerSearchUICommand {
+    func synchronizeAllLightCustomersDataAction(siteID: Int64, pageNumber: Int, pageSize: Int, onCompletion: ((Bool) -> Void)?) -> CustomerAction {
+        CustomerAction.synchronizeLightCustomersData(siteID: siteID, pageNumber: pageNumber, pageSize: pageSize) { result in
             switch result {
             case .success(_):
                 onCompletion?(result.isSuccess)
             case .failure(let error):
                 DDLogError("Customer Search Failure \(error)")
             }
         }
-        ServiceLocator.stores.dispatch(action)
-    }
-
-    func didSelectSearchResult(model: Customer, from viewController: UIViewController, reloadData: () -> Void, updateActionButton: () -> Void) {
-        onDidSelectSearchResult(model)
     }
 
-    func searchResultsPredicate(keyword: String) -> NSPredicate? {
-        return NSPredicate(format: "siteID == %lld AND ANY searchResults.keyword = %@", siteID, keyword)
+    func searchCustomersAction(siteID: Int64, keyword: String, pageNumber: Int, pageSize: Int, onCompletion: ((Bool) -> Void)?) -> CustomerAction {
+        CustomerAction.searchCustomers(siteID: siteID, keyword: keyword) { result in
+            switch result {
+            case .success(_):
```

Comment:
Non-blocker: Since we're not using the returning result value, we do not need to use the wildcard pattern:
```suggestion
            case .success:
```